_posts/2024-01-19-yannick-horrix-opentofu.md

```diff
@@ -0,0 +1,160 @@
+This abrupt change sent ripples across the Terraform community, sparking concerns about its future.
+
+In response, the open source community initiated the OpenTofu project, which was adopted by the Linux Foundation on September 20th. OpenTofu, starting off as a fork from Terraform 1.5.6, aims to create a truly open source alternative to Terraform. 
+
+On January 10, 2024, OpenTofu achieved a pivotal milestone by releasing its first production-stable version, OpenTofu 1.6.0, bringing new features and promising a seamless migration for current Terraform users.
+
+In this article we'll delve into a comparative analysis of OpenTofu and Terraform, exploring the impact of the license change, to help you make an informed decision when choosing the right tool for your infrastructure needs.
+# A Comparative Analysis: OpenTofu vs Terraform
+## Features
+In terms of functionality, both tools are currently equivalent. 
+The OpenTofu community has aimed to maintain feature parity with Terraform for the time being. 
+As such, both currently offer the same large set of features making them well suited tools for IaC workloads.
+
+Some small features were added with the release of OpenTofu 1.6.0 however, including:
+* An advanced testing feature for improved stability in configurations and modules
+* Enhanced S3 state backend with new authentication methods
+* A new provider and module registry, allowing a simpler method of publishing via pull request
+
+The upcoming version 1.7 aims to introduce more community requested features not available in Terraform, including:
+* Client-side state encryption for heightened security in regulated environments
+* Parameterizable backends, providers, and modules to enable more readable, DRY code
+* Third-party extensibility, with a plugin system for new state backends
+
+On Terraform's side, version 1.6 implemented performance improvements to the 'terraform test' command, as well as changes to the S3 backend.
+
+For now, both tools are functionally equivalent and compatible. 
+Time will tell whether this will remain the case as development continues on both sides.
+## Configuration and Syntax
+The core difference between OpenTofu and Terraform lies in its syntax. 
+OpenTofu uses a declarative syntax, where users can define their desired end state without having to worry about the steps required to get there. 
+Terraform, on the other hand, employs a procedural approach. 
+This means users have to define the specific steps, leading to higher control over the execution at the cost of higher complexity.
+
+In terms of configuration, OpenTofu focuses on simplicity in its configuration language, prioritizing readability and ease of use. 
+Terraform configuration, on the other hand, can be more verbose and complex to maintain.
+## Licensing and Pricing
+Both Terraform and OpenTofu are open-source tools, providing free usage.
```

Comment:
`Business Source License` (used by Terraform) is not an open-source license. It's a source-available license model https://fossa.com/blog/business-source-license-requirements-provisions-history/

Reply:
OpenTofu uses MPL 2.0 (in release 1.6.0)